src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -137,6 +138,59 @@ private string ResolveMatchedPath(string matchedPath)
             return RequestPathBase.Add(relativePath).ToString();
         }
 
+        private class PathComparer : IComparer<string>
+        {
+            public int Compare(string x, string y)
+            {
+                // < 0 = x < y
+                // > 0 = x > y
+
+                if (x.Equals(y))
```

Comment:
I'm actually interested on @Eilon thoughts on this, as this means the order of files will be culture sensitive (meaning current request culture). That doesn't seem *right* but the only fix would appear to be having a custom string comparer too.